### PR TITLE
Crypto compat cleanup

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
sigh

File: lib/crypto/crypto_compat.c [hunk 1]
```diff
@@ -213,3 +213,18 @@ crypto_compat_RSA_generate_key()
 	/* Failure! */
 	return (NULL);
 }
+
+/**
+ * crypto_compat_free(void):
+ * Free the shared memory allocated by OpenSSL.
+ */
+void
+crypto_compat_free(void)
```
Comment (gperciva) [inline lib/crypto/crypto_compat.c]: I'm not certain if you'd prefer `crypto_compat_cleanup()` here instead.  Most (if not all) of our other `_free` functions take a structure, but I figured that since this was a `(void)` it would be sufficiently clear.

**Comment by cperciva** [comment]
Does this DTRT with LibreSSL?  IIRC their versions map to OpenSSL versions in interesting ways.

**Comment by gperciva** [comment]
Yes-ish?  Empirically yes, theoretically yes, practically we can't rely on it working in the future (because that's just how SSL projects behave :( ).

It compiles and appears to work in:
- OpenBSD 6.1, which is libressl 2.5.3 (supports openssl 1.0.2g api)
- OpenBSD 6.4, which is libressl 2.8.2 (which added support for the opsnssl 1.1 api, but still includes the openssl 1.0.2g api)

That said, Alpine linux switched from libressl back to openssl early in 2018 [1].  Amongst their reasons, they claimed that "libressl promised to retain compatibility with 1.0.1g APIs, but has failed to do so.".  (I read the first portion of the large thread discussing this and didn't find specifics about this point, but neither did I find anybody disputing that claim.)

[1] https://lists.alpinelinux.org/alpine-devel/6073.html

At the moment, the OpenBSD libressl man pages show the expected functionality (and deprecation status) of `ERR_remove_state()` and `ERR_remove_thread_state()`:

https://man.openbsd.org/ERR_remove_state.3
https://man.openbsd.org/ERR_remove_thread_state.3

So in summary: I think we're good for now.  (Also, we have a separate issue for potential problems from libressl 2.7 onwards: https://github.com/Tarsnap/tarsnap/issues/319.  As far as I know, that's a theoretical problem so it's been low on my priority list.)

**Comment by cperciva** [comment]
Ok, just wanted to make sure you'd tested with LibreSSL.  Thanks!
